homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -83,83 +73,96 @@ def convert_and_round(
         key="charging_status",
         translation_key="charging_status",
         key_class="fuel_and_battery",
-        value=lambda x, y: x.value,
+        device_class=SensorDeviceClass.ENUM,
+        options=[s.value.lower() for s in ChargingState if s != ChargingState.UNKNOWN],
```

Comment:
@MartinHjelmare Usually we discourage developers to do this because "what if the library changes?", but in this case the options are encapsulated in the snapshots, so whenever the enum changes and the library is bumped (without regenerating the snapshots) tests would fail. So the only thing left is that the reviewer should know that a string should be added to the strings.

https://github.com/home-assistant/core/pull/110272/files#diff-3cb4a00db27894cb9e17559dd978b695621d9a673c6899905f94f839e97910ebR22-R35

Reply:
I still don't think it's a good idea to take that approach. The same reasoning still applies. The integration should blow up if an option is missing in the translations.

Reply:
But that can also happen if someone just adds a value to "a random list in const.py" without adding the transition key? It both comes down to the alertness of the reviewer on this. (And imo, with these snapshots it's even more clear what this change is and that a key should be added)

Reply:
> The integration should blow up if an option is missing in the translations.

I implemented a separate test only for the translations to make it clear after a library update if translations are missing (see latest commit).
Then users would not get error messages it is already part of the testing pipeline.

If a translation key is missing (or malformed), pytest assertion would throw an error similar to this:
```
>       assert all_sensor_options == all_translation_options
E       AssertionError: assert {'component.b...ot_full', ...} == {'component.b...ot_full', ...}
E         
E         Extra items in the left set:
E         'component.bmw_connected_drive.entity.sensor.charging_status.state.error'
E         Extra items in the right set:
E         'component.bmw_connected_drive.entity.sensor.charging_status.state.error1'
E         Use -v to get more diff
```

Reply:
Oh you added a full test for it, my point was more saying that having a way to fixate the options list somewhere would suffice instead of duplicating all enums.

Reply:
Wondering what I can do for this to go through?
Just pinning the enum sensors?

Reply:
I have reverted (i.e. pinned) the Enum values so this PR can be (hopefully) merged.

I might create another draft PR with a corresponding architecture discussion as I think that Enum values from a library should be allowed, provided there are tests if all translations are available.